### PR TITLE
Task/align estimated row count endpoint

### PR DESCRIPTION
**Pull request by esotrope**

### CODE REVIEW

File: webapp/src/main/webapp/src/app/report/report.service.ts [hunk 1]
```diff
@@ -192,16 +192,19 @@ export class ReportService {
...
   private toReport(remote: any): Report {
-    let local: Report = new Report();
+    const local: Report = new Report();
     local.id = remote.id;
     local.label = remote.label;
     local.status = remote.status;
     local.created = remote.created;
     local.reportType = remote.reportType;
     local.assessmentType = AssessmentType[ remote.assessmentType as string ];
+    // TODO report resource should hold the codes so we can remove this manual backend enum to code mapping
```
Comment (agorina) [inline webapp/src/main/webapp/src/app/report/report.service.ts]: Could you please clarify this TODO?

File: webapp/src/main/webapp/src/app/aggregate-report/results/aggregate-report.component.ts [hunk 2]
```diff
@@ -37,7 +38,8 @@ export class AggregateReportComponent implements OnInit, OnDestroy {
     this.assessmentDefinitionsByAssessmentTypeCode = this.route.snapshot.data[ 'assessmentDefinitionsByAssessmentTypeCode' ];
     this.options = this.route.parent.snapshot.data[ 'options' ];
     this.report = this.route.snapshot.data[ 'report' ];
-    this.reportSizeSupported = Number.parseInt(this.report.metadata.row_count) < SupportedRowCount;
+    this.reportSizeSupported = !Utils.isUndefined(this.report.metadata.row_count)
+      && (Number.parseInt(this.report.metadata.row_count) < SupportedRowCount);
```
Comment (esotrope) [inline webapp/src/main/webapp/src/app/aggregate-report/results/aggregate-report.component.ts]: makes it to where the report can be viewed if there is no row_count metadata stored with the report (not sure that piece is there yet)

File: webapp/src/main/webapp/src/app/report/report.service.ts [hunk 3]
```diff
@@ -192,16 +192,19 @@ export class ReportService {
...
+    local.assessmentTypeCode = {"MATH": "Math", "ELA": "ELA"}[remote.assessmentType];
     local.subjectId = AssessmentSubjectType[ remote.subject as string ] || 0;
+    local.subjectCode = {"ICA": "ica", "IAB": "iab", "SUMMATIVE": "sum"}[remote.subject];
     local.schoolYear = remote.schoolYear;
-    local.metadata = remote.metadata;
+    local.metadata = remote.metadata || {};
```
Comment (esotrope) [inline webapp/src/main/webapp/src/app/report/report.service.ts]: this is the change i went in to make. The rest is deprecation and a half step toward bridging the current with an eventual state

File: webapp/src/main/webapp/src/app/report/report.service.ts (hunk 1 above)
Comment (esotrope) [reply webapp/src/main/webapp/src/app/report/report.service.ts]: the API report resource model needs to be enhanced to return assessmentTypeCode, subjectCode (currently it returns java enum strings) Actually i can go in and fix this in this PR

**Comment by esotrope** [comment]
going to merge now to merge qa testing along

**Pull request merged**